backend/api/views.py

```
@@ -14,23 +15,30 @@
+    
+    def get_queryset(self):
+        # Regular users can only see their own profile
+        user = self.request.user
+        if user.is_staff:
+            return CustomUser.objects.all()
+        return CustomUser.objects.filter(user=user)
 
 class CompanyViewSet(viewsets.ModelViewSet):
     queryset = Company.objects.all()
     serializer_class = CompanySerializer
+    permission_classes = [IsAuthenticated]
```

Ruff (0.8.2)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)